ansys/tools/repo_sync/repo_sync.py

```diff
@@ -90,53 +63,27 @@ def synchronize(
         )
 
         # unsafe, should add specific file or directory
-        process = subprocess.Popen(
-            ["git", "add", "--a"],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-        stdout, stderr = process.communicate()
+        subprocess.check_call(["git", "add", "--a"], cwd=repo_path)
 
         if protos_path:
             message = f"""Add folder content from {protos_path}."""
         else:
             message = f"Copy all files located into the {repository} repository from branch {branch_name}."
 
         if dry_run:
-            process = subprocess.Popen(
-                ["git", "commit", "-am", message, "--dry-run"],
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-            )
-            stdout, stderr = process.communicate()
+            output = subprocess.check_output(["git", "commit", "-am", message, "--dry-run"], cwd=repo_path)
             print("Dry-run synchronization output:")
-            print(stdout)
+            print(output)
         else:
-            process = subprocess.Popen(
-                ["git", "commit", "-am", message],
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-            )
-            stdout, stderr = process.communicate()
-
-            process = subprocess.Popen(
-                ["git", "push", "-u", "origin", branch_name],
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-            )
-            stdout, stderr = process.communicate()
+            subprocess.check_call(["git", "commit", "-am", message], cwd=repo_path)
+
+            subprocess.check_call(["git", "push", "-u", "origin", branch_name], cwd=repo_path)
 
             # Create pull request.
             gh = github.Github(token)
             repo = gh.get_repo(f"{organization}/{repository}")
             pr = repo.create_pull(title=message, body=message, head=branch_name, base="main")
 
-        # Delete the git repository that was created.
-        parent_folder = os.path.dirname(os.getcwd())
-        os.chdir(parent_folder)
-        shutil.rmtree(os.path.join(parent_folder, repository), onerror=_on_rm_error)
-        os.chdir(os.path.dirname(os.getcwd()))
-
```

Comment:
This is not needed, because the `tempfile.TemporaryDirectory` cleans up after itself.